c++/src/kj/compat/http.c++

```diff
@@ -2853,9 +2853,16 @@ public:
             }
             bool addNullTerminator = true;
             // We want to add the null terminator when receiving a TEXT message.
-            auto decompressed = decompressor.processMessage(message, originalMaxSize,
-                addNullTerminator);
-            return Message(kj::String(decompressed.releaseAsChars()));
+            auto decompressedOrError = decompressor.processMessage(message, originalMaxSize, addNullTerminator);
+            KJ_SWITCH_ONEOF(decompressedOrError) {
+              KJ_CASE_ONEOF(protocolError, ProtocolError) {
+                KJ_DBG("Returning error with", protocolError.description);
+                return sendCloseDueToError(protocolError.statusCode, protocolError.description).attach(kj::mv(decompressedOrError));
```

Comment:
Nitpick: [KJ style guide](https://github.com/capnproto/capnproto/blob/v2/style-guide.md) requires wrapping at 100 columns.

```suggestion
                return sendCloseDueToError(protocolError.statusCode, protocolError.description)
                    .attach(kj::mv(decompressedOrError));
```

Can you fix the other places too? I suggest configuring your editor to show a line at 100 columns.